label_studio/projects/functions/next_task.py

```diff
@@ -13,30 +13,13 @@
 
 
 def _get_random_unlocked(task_query, user, upper_limit=None):
-    # get random task from task query, ignoring locked tasks
-    n = task_query.count()
-    if n > 0:
-        upper_limit = upper_limit or n
-        random_indices = np.random.permutation(upper_limit)
-        task_query_only = task_query.only('overlap', 'id')
-
-        for i in random_indices:
-            try:
-                task = task_query_only[int(i)]
-            except IndexError as exc:
-                logger.error(
-                    f'Task query out of range for {int(i)}, count={task_query_only.count()}. ' f'Reason: {exc}',
-                    exc_info=True,
-                )
-            except Exception as exc:
-                logger.error(exc, exc_info=True)
-            else:
-                try:
-                    task = Task.objects.select_for_update(skip_locked=True).get(pk=task.id)
-                    if not task.has_lock(user):
-                        return task
-                except Task.DoesNotExist:
-                    logger.debug('Task with id {} locked'.format(task.id))
+    for task in task_query.order_by('?').only('id')[:settings.RANDOM_NEXT_TASK_SAMPLE_SIZE]:
```

Comment:
are you sure we can skip 'overlap'?

Reply:
Looks like we call `has_lock` on the task object with all field: 
```
                    task = Task.objects.select_for_update(skip_locked=True).get(pk=task.id)
                    if not task.has_lock(user):
```